crates/typst-layout/src/inline/shaping.rs

```diff
@@ -932,6 +961,65 @@ fn shape_segment<'a>(
     ctx.used.pop();
 }
 
+/// Returns a `(script_shift, script_compensation, scale)` triplet describing
```

Comment:
The doc comment is outdated. The function name doesn't seem 100% correct anymore since it also produces the feature. How about `determine_shift`? Or splitting it into two functions `compute_synthesized_shift(font: &Font, settings: ShiftSettings) -> (Em, Em, Em)` and `shift_feature(text: &str, font: &Font, settings: ShiftSettings) -> Option<Feature>` and then tying it together above.